### PR TITLE
new leaderboard features

### PR DESCRIPTION
**Pull request by scherna**

### CODE REVIEW

File: app/containers/Leaderboard/Leaderboard.js [hunk 1]
```diff
@@ -20,11 +20,69 @@ export const Leaderboard = React.createClass({
...
+			leaderSort: 'byStateCount',
+		};
+	},
+
+    leadersByStateCount() {
+        const leaders = this.props.leaderboardData.leaders || [];
+        const flatLeaders = flattenLeaders(leaders).sort(function (a,b) {
+            return cmp(a.statesCount, b.statesCount) || cmp(a.score, b.score);
+        }).reverse();
+        console.log("called");
```
Comment (jasrub) [inline app/containers/Leaderboard/Leaderboard.js]: remove console.log calls, they might be confusing for later development. (also unnecessary for users to see)

File: app/containers/Leaderboard/Leaderboard.js [hunk 2]
```diff
@@ -20,11 +20,69 @@ export const Leaderboard = React.createClass({
...
+	},
+
+    toggleByNumChildren: function() {
+		this.setState({ leaderSort: 'byNumChildren' });
+	},
+
+	render() {
+        let leaders = [];
+        const myId = JSON.parse(localStorage.getItem('userData'))["id"];
+        console.log(myId);
```
Comment (jasrub) [inline app/containers/Leaderboard/Leaderboard.js]: another console.log to remove

File: app/containers/Leaderboard/Leaderboard.js [hunk 3]
```diff
@@ -20,11 +20,69 @@ export const Leaderboard = React.createClass({
...
+            leaders = this.leadersByStateCount();
+        }
+        else if (this.state.leaderSort === 'byScore') {
+            leaders = this.leadersByScore();
+        }
+        else if (this.state.leaderSort === 'byNumChildren') {
+            leaders = this.leadersByNumChildren();
+        }
+        const myLeaderIndex = leaders.findIndex(function (leader) {
+            console.log(leader.user.id);
```
Comment (jasrub) [inline app/containers/Leaderboard/Leaderboard.js]: console.log

File: config.js [hunk 4]
```diff
@@ -1,3 +1,5 @@
-// process.env.API_URL = 'http://localhost:9876';
-process.env.API_URL = 'https://dev.fiftynifty.org/api';
+
```
Comment (jasrub) [inline config.js]: I'll add config.js to .gitignore
no need to push local changes to it with each commit.

In the meantime, can you remove this file from the pull request?

File: app/containers/User/actions.js [hunk 5]
```diff
@@ -85,3 +90,28 @@ export function requestLatLong(address, userId) {
 		});
 	};
 }
+
+export function putUserUpdate(userId, name, zipcode) {
+	return (dispatch) => {
+		dispatch({ type: POST_USER_UPDATE_LOAD });
+		return clientFetch('/api/user/update', {
```
Comment (jasrub) [inline app/containers/User/actions.js]: Is this endpoint already in our api?
If not - let's push the api change first

**Comment by scherna** [comment]
the "yarn.lock" that you mentioned is something from npm I believe, but I'm not sure. I believe it came into my branch after pulling from Jason, so I imagine he added it at some point

**Pull request closed without merging**